icgcget/cli.py

```diff
@@ -112,90 +113,93 @@ def cli(ctx, config, logfile):
+        if gdc_access is None:
+            logger.error("No credentials provided for the gdc repository.")
+            raise click.BadParameter
+
+    for repository in repo:
+        object_ids[repository] = []
+
+    for fileid in fileids:
+        info = icgc_api.check_file(fileid)
+        if type(info) is int:
+            raise click.ClickException("File not found")
+        repository, copy = match_repositories(repo, info)
+        if repository is None:
+            logger.error("File {} specified not found on repositories {}".format(info["id"], repo))
+        object_ids[repository].append(copy["indexFile"]["objectId"])
+
+    if 'aws-virginia' in repo:
+        if 'aws-virginia' in object_ids and object_ids['aws-virginia']:
+            if len(object_ids['aws-virginia']) > 1:  # Todo-find a workaround for this: dynamic manifest generation?
+                logger.error("The icgc repository does not support input of multiple file id values.")
+                raise click.BadParameter
+            code = icgc_client.icgc_call(object_ids['aws-virginia'], icgc_access, icgc_path, icgc_transport_file_from,
+                                         icgc_transport_parallel, output, 'aws')
+
+    if 'ega' in repo:
+        if object_ids['ega']:
+            if len(object_ids['ega']) > 1:  # Todo-find a workaround for this rather than throw errors
                 logger.error("The ega repository does not support input of multiple file id values.")
-                code = 1
-                return code
+                raise click.BadParameter
             else:
                 if ega_transport_parallel != '1':
                     logger.warning("Parallel streams on the ega client may cause reliability issues and failed " +
                                    "downloads.  This option is not recommended.")
-                code = ega_client.ega_call(fileid, ega_username, ega_password, ega_path, ega_transport_parallel,
-                                           ega_udt, output)
+                code = ega_client.ega_call(object_ids['ega'], ega_username, ega_password, ega_path,
+                                           ega_transport_parallel, ega_udt, output)
                 if code != 0:
-                    logger.error("{} exited with a nonzero error code.".format(repo))
+                    logger.error("client exited with a nonzero error code {}.".format(code))
+                    click.ClickException("Please check client output for error messages")
 
-        if manifest is True:
-            logger.warning(
-                "The ega repository doesn't support downloading from manifest files. Use the -f tag instead")
-            code = 1
-            return code
+    if 'cghub' in repo:
+        if object_ids['cghub']:
+            code = gt_client.genetorrent_call(object_ids['cghub'], cghub_access, cghub_path,
+                                              cghub_transport_parallel, output)
+            if code != 0:
```

Comment:
Might want to add a `check_code` function that does these 3 steps. Encapsulates the repeating parts and removes the need for a branch. I would also pass in some additional context information like the repo and maybe the client call.